agent/apps/ai_guardrails/app_code/permissions.py

```
@@ -0,0 +1,29 @@
+Custom Permissions
+"""
+
+from app_standards.views.permissions import BasePermission
+
+SAFE_METHODS = {"GET", "HEAD", "OPTIONS"}
+
+class IsGuardrailsAuthenticated(BasePermission):
+    """
+    فقط کاربر لاگین‌کرده اجازه دسترسی داشته باشه.
+    """
+    def has_permission(self, request, view) -> bool:
```

Ruff (0.12.2)

Missing type annotation for function argument `request`

(ANN001)